src/main/kotlin/io/github/anki/anki/controller/AuthController.kt

```diff
@@ -6,57 +6,59 @@ import io.github.anki.anki.controller.dto.auth.SignUpRequestDto
 import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Mono
 
 @CrossOrigin(
     origins = ["*"],
     maxAge = AuthController.MAX_AGE.toLong(),
 )
 @RestController
 @RequestMapping(AuthController.BASE_URL)
-class AuthController @Autowired constructor(
-    val userService: UserService,
-    val secureService: SecurityService,
+class AuthController(
+    private val userService: UserService,
+    private val encoder: PasswordEncoder,
 ) {
 
     @PostMapping(SIGN_IN)
     @ResponseStatus(HttpStatus.OK)
-    fun authenticateUser(@RequestBody signInRequestDto: @Valid SignInRequestDto?): JwtResponseDto {
-        LOG.info(
-            "IN: ${AuthController::class.java.name}: ${BASE_URL}${SIGN_IN} with userName${signInRequestDto!!.userName}",
-        )
-
+    fun authenticateUser(@RequestBody signInRequestDto: @Valid SignInRequestDto): Mono<JwtResponseDto> =
         userService.signIn(signInRequestDto.toUser())
-        val authentication = secureService.authUser(signInRequestDto)
-        val authUser = authentication.principal as User
-
-        LOG.info("OUT: ${AuthController::class.java.name}: ${BASE_URL}${SIGN_IN} with ${HttpStatus.OK}")
-        return authUser.toJwtDto(secureService.jwtUtils.generateJwtToken(authentication))
-    }
+            .doFirst {
+                LOG.info(
+                    "IN: ${AuthController::class.java.name}: ${BASE_URL}${SIGN_IN} with userName {}",
+                    signInRequestDto.userName,
+                )
+            }
+            .map(UserAuthentication::toJwtDto)
+            .doOnNext {
+                LOG.info("OUT: ${AuthController::class.java.name}: $BASE_URL$SIGN_IN with ${HttpStatus.OK}")
+            }
 
     @PostMapping(SIGN_UP)
     @ResponseStatus(HttpStatus.CREATED)
-    fun registerUser(@RequestBody signUpRequestDto: @Valid SignUpRequestDto): UserCreatedMessageResponseDto {
-        LOG.info("IN: ${AuthController::class.java.name}: ${BASE_URL}${SIGN_UP} with $signUpRequestDto")
-
-        userService.signUp(signUpRequestDto.toUser(secureService.encoder.encode(signUpRequestDto.password)))
-
-        LOG.info("OUT: ${AuthController::class.java.name}: ${BASE_URL}${SIGN_IN} with ${HttpStatus.OK}")
-        return UserCreatedMessageResponseDto(CREATED_USER_MESSAGE)
-    }
+    fun registerUser(@RequestBody signUpRequestDto: @Valid SignUpRequestDto): Mono<UserCreatedMessageResponseDto> =
+        userService.signUp(signUpRequestDto.toUser(encoder))
+            .doFirst {
+                LOG.info(
+                    "IN: ${AuthController::class.java.name}: $BASE_URL$SIGN_UP with {}",
+                    signUpRequestDto,
+                )
+            }
+            .map { UserCreatedMessageResponseDto(CREATED_USER_MESSAGE) }
+            .doOnNext { LOG.info("OUT: ${AuthController::class.java.name}: $BASE_URL$SIGN_IN with ${HttpStatus.OK}") }
```

Comment:
this log is useless. It doesn't have the relevant context. 

Suppose you have a multiple parallel requests. Then how you know which one request triggered this log?

the log in `doFirst` is ok, because it has `signUpRequestDto`